### PR TITLE
Hestia: Up-Armored Edition.

### PR DESCRIPTION
**Pull request by Dark1Star3s**
The Great Leader is Dead.

## Changelog

- Mech Armor Fixed and now applies.
- Mech Armor Buffed across the board, all armor is stronger and more resilient.
- Mech Health buffed a lot. Minimum health on the lightest mechs from 70 to 300. The Highest at 500.

### CODE REVIEW

**Comment by githubuser4141** [comment]
Cool, but.. WTF's with the limb balance? Light components shouldn't be as protective as combat ones, and they shouldn't be multiple times tougher then power-loader components surely. The flavour desc. suggests Light parts are performance-oriented, so think of a sports car vs. a utility truck (power-loader) vs. a up-armored HMMWV (combat parts) vs. a... I'm not a tank buff, but let's say a Stryker-like armored vehicle (heavy parts).

At the end of the day, a mech isn't that much bigger than a human. It's small enough to fit through an airlock and can use the same hallways and staircases as humans. It's individual components can be picked up and carried by human hands suggesting it's not actually as heavy or giant as you'd think. Most likely it's just a very up-sized RIG that doesn't have articulated limbs. That's also not accounting for weight and bulkiness that's 'wasted' to make up the mech's space-proofness, heat-proofness and onboard life/temperature/oxygen systems, stuff that isn't necessary on IRL plate carriers armored vehicles. So it's somewhat reasonable that it'd be vulnerable to infantry carried rifles.

And.. you do realise EM armour is supposed to be less protective against standard ammunition because it makes the mech 100% immune to EMPs, right? It might not be 100% realistic for a heavily armoured vehicle to be vulnerable to pistol rounds but /realistically/ there's a heap of issues with armoured vehicles that can't/won't be simulated here so having a trade-off between ballistic/melee/laser resistance and EMP-proofness is probably good. 

Also, material costs should probably be increased otherwise you'll be almost be out-armouring and arming the entirety of security round-start with just a set of combat parts which require only iron and glass. Maybe do what ERIS did and add 
uranium/diamond/plasma requirements to combat/heavy components so it's not totally lopsided.

**Comment by Dark1Star3s** [comment]
> Cool, but.. WTF's with the limb balance? Light components shouldn't be as protective as combat ones, and they shouldn't be multiple times tougher then power-loader components surely. The flavour desc. suggests Light parts are performance-oriented, so think of a sports car vs. a utility truck (power-loader) vs. a up-armored HMMWV (combat parts) vs. a... I'm not a tank buff, but let's say a Stryker-like armored vehicle (heavy parts).

Power Loaders are open to space, their entire chassis is open to the elements- while they are weaker, they hit harder than the light mech. On top of that I will be adding more material costs to these parts later, I would like to see how these handle in actual combat first. To add to your second point, EM armor is shielded against EM, it is just "Resistant" which is pretty low in armor ratings for Hestia, depending which chassis it's mounted to, you probably could kill it with just 1 or 2 mags from a bulldog.

Research is also a factor, and Robotics will require research to do their thing- and, command shouldn't be authorising battlemechs to be built- think of it like the armory. Security are not allowed their giga-chad guns until red alert. If Command starts letting Robotics build battlemechs and they start going around wrecking the place that is on them. But it's also secs job to alert command. All in all I am testing the armor buffs first, I want to see what people think of them for a while, this also fixes mech armor not working in the first place.

Purple and I spoke heavily on the balance of these and I believe it's at a good point to begin testing.

**Comment by githubuser4141** [comment]
> > Cool, but.. WTF's with the limb balance? Light components shouldn't be as protective as combat ones, and they shouldn't be multiple times tougher then power-loader components surely. The flavour desc. suggests Light parts are performance-oriented, so think of a sports car vs. a utility truck (power-loader) vs. a up-armored HMMWV (combat parts) vs. a... I'm not a tank buff, but let's say a Stryker-like armored vehicle (heavy parts).
> 
> Power Loaders are open to space, their entire chassis is open to the elements- while they are weaker, they hit harder than the light mech. On top of that I will be adding more material costs to these parts later, I would like to see how these handle in actual combat first. To add to your second point, EM armor is shielded against EM, it is just "Resistant" which is pretty low in armor ratings for Hestia, depending which chassis it's mounted to, you probably could kill it with just 1 or 2 mags from a bulldog.
> 
> Research is also a factor, and Robotics will require research to do their thing- and, command shouldn't be authorising battlemechs to be built- think of it like the armory. Security are not allowed their giga-chad guns until red alert. If Command starts letting Robotics build battlemechs and they start going around wrecking the place that is on them. But it's also secs job to alert command. All in all I am testing the armor buffs first, I want to see what people think of them for a while, this also fixes mech armor not working in the first place.
> 
> Purple and I spoke heavily on the balance of these and I believe it's at a good point to begin testing.

Your changes don't reflect PL arms hitting harder, so my bad. I think however while the chassis is 'open' lore-wise it's still supposed to a tough steel cage vs. the light chassis which is a sealed plastic and glass can, obviously the pilots are going take the brunt of the hits to PL chests, but the parts themselves are supposed to be tougher.

As for research/combat mechs, roboticists can finish research in about 30 to 40 minutes if they chose, and.. this isn't exactly TG where mechs are non-modular. Mech armor isn't visible, and you gave basic non-combat armor combat stats, so you could say an Odysseus is now a combat mech durability-wise. Also, 'heavy' parts which are absolutely useful in combat scenarios aren't actually named as such, so someone could build a "mining mech" with heavy parts, combat armor and who's going to stop them if they don't add actual guns? Plasma cutters which are a "utility" tool could be added then you'd have a 1200 HP validhunting suit that's almost invulnerable to anything that's not an EMP. In a localhost server, laser rifles did about 3 damage to a mech w/ heavy armor, and basic mechs are effectively immune to melee. Unless the pilot is AFK, you effectively cannot fight them without an ion weapon.

**Comment by Dark1Star3s** [comment]
> > > Cool, but.. WTF's with the limb balance? Light components shouldn't be as protective as combat ones, and they shouldn't be multiple times tougher then power-loader components surely. The flavour desc. suggests Light parts are performance-oriented, so think of a sports car vs. a utility truck (power-loader) vs. a up-armored HMMWV (combat parts) vs. a... I'm not a tank buff, but let's say a Stryker-like armored vehicle (heavy parts).
> > 
> > 
> > Power Loaders are open to space, their entire chassis is open to the elements- while they are weaker, they hit harder than the light mech. On top of that I will be adding more material costs to these parts later, I would like to see how these handle in actual combat first. To add to your second point, EM armor is shielded against EM, it is just "Resistant" which is pretty low in armor ratings for Hestia, depending which chassis it's mounted to, you probably could kill it with just 1 or 2 mags from a bulldog.
> > Research is also a factor, and Robotics will require research to do their thing- and, command shouldn't be authorising battlemechs to be built- think of it like the armory. Security are not allowed their giga-chad guns until red alert. If Command starts letting Robotics build battlemechs and they start going around wrecking the place that is on them. But it's also secs job to alert command. All in all I am testing the armor buffs first, I want to see what people think of them for a while, this also fixes mech armor not working in the first place.
> > Purple and I spoke heavily on the balance of these and I believe it's at a good point to begin testing.
> 
> Your changes don't reflect PL arms hitting harder, so my bad. I think however while the chassis is 'open' lore-wise it's still supposed to a tough steel cage vs. the light chassis which is a sealed plastic and glass can, obviously the pilots are going take the brunt of the hits to PL chests, but the parts themselves are supposed to be tougher.
> 
> As for research/combat mechs, roboticists can finish research in about 30 to 40 minutes if they chose, and.. this isn't exactly TG where mechs are non-modular. Mech armor isn't visible, and you gave basic non-combat armor combat stats, so you could say an Odysseus is now a combat mech durability-wise. Also, 'heavy' parts which are absolutely useful in combat scenarios aren't actually named as such, so someone could build a "mining mech" with heavy parts, combat armor and who's going to stop them if they don't add actual guns? Plasma cutters which are a "utility" tool could be added then you'd have a 1200 HP validhunting suit that's almost invulnerable to anything that's not an EMP. In a localhost server, laser rifles did about 3 damage to a mech w/ heavy armor, and basic mechs are effectively immune to melee. Unless the pilot is AFK, you effectively cannot fight them without an ion weapon.

I get your points, though you shouldn't be fighting a mech 1v1. It's like fighting a tank while you're a single soldier in the open. Dagon has 2 Ion Rifles and 2 Ion Pistols. Each one in the other armory. So if Ions become an issue that's securities fault. But I can see your points. I'll have to talk to purple later to see how we can make it to where combat armor is visible on the mech.

**Comment by Dark1Star3s** [comment]
To add to that, once the main body is destroyed, (about 300 in the lightest body) armor is no longer effective and weapons do full damage again.

**Comment by Anton-Kr** [comment]
5 incorrect indentations (expecting exactly 0)

**Review by Anton-Kr — CHANGES_REQUESTED** [review]
fix CI

**Comment by Dark1Star3s** [comment]
I have no idea indentations are. :(

File: code/modules/mechs/mech_damage.dm [hunk 1]
```diff
@@ -28,11 +28,11 @@
 
 
 /mob/living/exosuit/get_armors_by_zone(def_zone, damage_type, damage_flags)
-	. = ..()
-	if(body && body.armor)
-		var/body_armor = get_extension(body.armor, /datum/extension/armor)
-		if(body_armor)
-			. += body_armor
+    . = ..()
```
Comment (Anton-Kr) [inline code/modules/mechs/mech_damage.dm]: this should be tabs

**Review by Anton-Kr — CHANGES_REQUESTED** [review]

**Comment by Dark1Star3s** [comment]
Okay. I think that did it.

**Comment by Dark1Star3s** [comment]
Not entirely sure why the map checks are failing. It doesn't change anything...

**Comment by Anton-Kr** [comment]
Map checks shouldn't be related to your code, just CI randomly fucking up. I'll try to get to this tmrw or in a few days. Ping me if I don't.

**Pull request merged**